File: Python/dos_quad.py
import numpy as np
def dos_quad(n,tn):
    tn = tn.reshape(len(tn),1)
    tj = np.linspace(0,2*n-1,2*n)*np.pi/n
    tj = tj.reshape(len(tj),1)
    m = np.linspace(1,n-1,n-1)
    m = m.reshape(1,len(m))
    R = np.zeros((tj.size,tj.size))
    for ii in range(len(tn)):
        ti = tn[ii]
        M = np.cos(m*(ti-tj))/np.tile(m,np.shape(tj))
        M = M.T
        temp = np.pi/n**2*np.cos(n*(ti-tj))
        temp = temp.reshape(len(temp))
        R[ii,:] = -2*np.pi/n*np.sum(M,axis = 0)-temp
    return R

File: Matlab/dos_quad.m
function R = dos_quad(n, tn)
%反散射93页R_j^(n)公式
    s = size(tn);
    if s(2) > s(1)
        tn = tn';
    end
    tj = (0:(2*n-1))*pi/n;
    m = (1:(n-1))';
    R = zeros(length(tn), length(tj));
    for i = 1:length(tn)
        ti = tn(i);
        M = cos(m*(ti - tj)) ./ repmat(m, size(tj));
        R(i,:) = -2*pi/n*sum(M, 1) - pi/n^2 * cos(n*(ti - tj));
    end
    
end